### ISSUE TITLE
Mocking waitFor calls

### ISSUE THREAD

**Issue by m-radzikowski** [m-radzikowski/aws-sdk-client-mock#12]
See https://github.com/aws/aws-sdk-js-v3/issues/2215

**Comment by evanstachowiak** [comment]
@m-radzikowski any update on this?

I'm trying to mock cloudfront `waitUntilInvalidationCompleted`. I think this would helpful.

I also see there is the `callsFake` method but I haven't figured out how to get that working, if that's even the way to go. Any suggestions would be greatly appreciated.

**Comment by m-radzikowski** [comment]
No, it's not supported directly by the library. The best approach is to mock the `waitUntilInvalidationCompleted` function with Jest or other lib, like this: https://jestjs.io/docs/mock-functions#mocking-partials

**Comment by evanstachowiak** [comment]
Thanks for the quick response! I managed to mock successfully as you suggested.

**Comment by evanstachowiak** [comment]
For future reference, the workaround I found:
```
jest.mock('@aws-sdk/client-cloudfront', () => ({
  ...jest.requireActual('@aws-sdk/client-cloudfront'),
  waitUntilInvalidationCompleted: jest.fn(async () => cloudfrontWaiterResponse),
}));
```
